packages/shared/invokeGuardedCallbackImpl.js

```diff
@@ -7,26 +7,25 @@
  * @flow
  */
 
-function invokeGuardedCallbackProd<A, B, C, D, E, F, Context>(
+function invokeGuardedCallbackProd<Args: Array<mixed>, Context>(
```

Comment:
Is this change required for well-formed-exports? Looks to me like we're giving up type-safety here?

Reply:
Noticed this is part of another PR. It's the bottom commit I need to look at :D

Reply:
Shouldn't be a loss of type safety as commented on another PR now.

I also untangled the PRs and made them into parallel PRs which seem to be better for review.